pkg/registry/client.go

```diff
@@ -105,11 +105,13 @@ func NewClient(options ...ClientOption) (*Client, error) {
 			if err != nil {
 				return nil, errors.New("unable to retrieve credentials")
 			}
-			// A blank returned username and password value is a bearer token
-			if username == "" && password != "" {
-				headers.Set("Authorization", fmt.Sprintf("Bearer %s", password))
-			} else {
-				headers.Set("Authorization", fmt.Sprintf("Basic %s", basicAuth(username, password)))
+			if username != "" || password != "" {
```

Comment:
To me, this logic could be most plainly written as:

```
if password != "" {
    if username != "" {
        headers.Set("Authorization", fmt.Sprintf("Basic %s", basicAuth(username, password)))
    } else {
       // Without a username, we assume this is a bearer token.
       headers.Set("Authorization", fmt.Sprintf("Bearer %s", password))
    }
}
```

(but I am nit-picking, and fixing this bug is priority)